### PR TITLE
Feature: basic implementation v0.1.0

### PR DESCRIPTION
**Pull request by vadymmarkov**
@hyperoslo/ios 
Sorry for a huge PR, but it was needed to make some experiments, try different things and actually start building something that works 😃 

**The most important features introduced here**:
- Support [S4](https://github.com/open-swift/S4) HTTP standards for Swift, so `Flamingo` could be run on any S4 supporting server. 
- Built-in [VeniceX](https://github.com/VeniceX/Venice) based [HTTP server](https://github.com/VeniceX/HTTPServer.git).
- New routing system.
- Middleware for parameter parsing and error handling.
- View rendering with [Stencil](https://github.com/kylef/Stencil).
- Everything is covered with tests, and `Quick` is replaced by `XCTest`.
- Code is well documented I believe, and Readme has only basic stuff while the project is under continuous development. 

There is a `Development` target, so if you follow the steps from Readme and build the package, the server could be run by `./.build/debug/Development`. Then just open `http://0.0.0.0:8000/` in your browser.

Public API looks like this at the moment, but I'm pretty sure it will be changed many times:

``` swift
import Flamingo

Config.viewsDirectory = "Sources/Development/Views"

let controller = Controller()

Flamingo.application.router.draw { map in
  map.root(respond: controller.index)
  map.resources("users", controller: Controller.self)

  map.namespace("api") { map in
    map.fallback(respond: controller.index)
    map.root(respond: controller.index)
    map.resources("users", controller: Controller.self)
    map.get("log", respond: controller.index)
  }
}

try Flamingo.application.start()
```

The next step is to make a lot of GitHub issues, so anyone can pick up and contribute. All smart ideas an pull requests are welcome!

### CODE REVIEW

**Comment by vadymmarkov** [comment]
@zenangst I think I need a help with Travis.

**Comment by vadymmarkov** [comment]
And btw

screen shot 2016-06-02 at 09 22 55

**Comment by vadymmarkov** [comment]
So Travis is finally fixed both for Mac and Linux. Thanks @zenangst. So what do you guys think @zenangst @RamonGilabert @onmyway133? If it's merged there are a lot of issues I've created https://github.com/hyperoslo/Flamingo/issues. Anyone is up to pick one and contribute? 😄

**Comment by onmyway133** [comment]
@vadymmarkov You rock 🎸 How could you do this

File: Sources/Flamingo/Core/Application.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  // MARK: - Server
+
+  /**
+    Starts HTTP server with the router as a responder.
+    - Throws: ErrorType.
+  */
+  public func start() throws {
+    print(Art.header)
```
Comment (zenangst) [inline Sources/Flamingo/Core/Application.swift:42]: Should we include a `print` here?

Comment (vadymmarkov) [inline Sources/Flamingo/Core/Application.swift:42]: What do you mean? Here we print ASCII art to the console.

Comment (zenangst) [inline Sources/Flamingo/Core/Application.swift:42]: Oh my bad, I just saw `print` and went 🤔

Comment (vadymmarkov) [inline Sources/Flamingo/Core/Application.swift:42]: 😄

**Comment by zenangst** [comment]
This PR is a bit overwhelming, went through it and couldn't really find anything to pick on. I really like it, terrific job @vadymmarkov !

brilliant

**Comment by vadymmarkov** [comment]
Next time it's gonna be not so huge, I promise... 😄

**Pull request merged**